compiler/src/iree/compiler/Codegen/Common/GPU/GPULowerToUKernels.cpp

```diff
@@ -54,9 +132,21 @@ static FailureOr<IREE::Codegen::UKernelOpInterface>
 matchArgmaxDAGForUKernel(RewriterBase &rewriter, linalg::GenericOp op) {
   auto targetAttr = IREE::HAL::ExecutableTargetAttr::lookup(op);
   const char ukernelName[] = "argmax";
-  if (!hasUkernel(targetAttr, ukernelName) ||
-      !hasUkernelSupportedGpuArch(targetAttr)) {
-    return failure();
+  Value input = op.getDpsInputOperand(0)->get();
+  auto inputType = cast<ShapedType>(input.getType());
+  Value index = op.getDpsInitOperand(1)->get();
+  auto indexType = cast<ShapedType>(index.getType());
+  std::string suffix;
+  llvm::raw_string_ostream(suffix)
+      << inputType.getElementType() << indexType.getElementType();
+  FnNameAndDefAttrs fn =
+      getFnNameAndDefAttrs(ukernelName, suffix, rewriter, targetAttr);
+  if (!fn) {
+    return rewriter.notifyMatchFailure(op, "No ukernels on this backend.");
```

Comment:
error messages are prefixed/suffixed by the printer and so have the form of "no ukernels on this backend" (so start lowercase and don't include trailing punctuation) - here and elsewhere
e.g.
![image](https://github.com/user-attachments/assets/a7a712eb-3d1d-44ac-8f06-2f7de36dd2dd)